### PR TITLE
seperate people_recognition from image_recognition

### PR DESCRIPTION
**Pull request by MatthijsBurgh**

### CODE REVIEW

File: launch/state_machines/challenge_help_me_carry.launch [hunk 1]
```diff
@@ -2,27 +2,40 @@
...
+            <arg name="machine" value="amigo2"/>
+            <arg name="config" value="world_modeling/world_model_help_me_carry.yaml"/>
+        </include>
 
-    		<!-- WORLD MODEL -->
-    		<include file="$(find robot_launch_files)/launch/world_modeling/world_model.launch">
-    			<arg name="machine" value="amigo2"/>
-    			<arg name="config" value="world_modeling/world_model_help_me_carry.yaml"/>
-    		</include>
+        <!-- IMAGE RECOGNITION -->
```
Comment (PetervDooren) [inline launch/state_machines/challenge_help_me_carry.launch:25]: Is image recognition needed in help me carry?

File: launch/state_machines/challenge_open.launch [hunk 2]
```diff
@@ -2,7 +2,9 @@
 
 <launch>
 
-    <group ns="amigo">
+    <arg name="robot_name" value="amigo" />
+
+    <group ns="$(arg robot_name)">
```
Comment (PetervDooren) [inline launch/state_machines/challenge_open.launch:7]: I think people and image recognition might be needed here

File: launch/state_machines/challenge_set_a_table.launch [hunk 3]
```diff
@@ -19,6 +21,17 @@
             <arg name="machine" value="amigo2"/>
         </include>
 
+        <!-- IMAGE RECOGNITION -->
+        <include file="$(find robot_launch_files)/launch/world_modeling/image_recognition.launch">
+            <arg name="machine" value="amigo2"/>
+        </include>
+
+        <!-- PEOPLE RECOGNITION -->
```
Comment (PetervDooren) [inline launch/state_machines/challenge_set_a_table.launch:29]: Is people recognition needed here?

File: launch/state_machines/challenge_spr.launch [hunk 4]
```diff
@@ -2,27 +2,40 @@
...
-        </group>
+        <!-- WORLD MODEL -->
+        <include file="$(find robot_launch_files)/launch/world_modeling/world_model.launch">
+            <arg name="machine" value="amigo2"/>
+            <arg name="config" value="world_modeling/world_model_spr.yaml"/>
+        </include>
 
-        <!-- Start the challenge -->
-        <node pkg="challenge_spr" type="challenge_spr.py" name="state_machine" args="amigo" output="screen" required="true"/>
+        <!-- IMAGE RECOGNITION -->
```
Comment (PetervDooren) [inline launch/state_machines/challenge_spr.launch:25]: Image recognition is not needed in spr

File: launch/state_machines/challenge_storing_groceries.launch [hunk 5]
```diff
@@ -19,6 +21,17 @@
             <arg name="machine" value="amigo2"/>
         </include>
 
+        <!-- IMAGE RECOGNITION -->
+        <include file="$(find robot_launch_files)/launch/world_modeling/image_recognition.launch">
+            <arg name="machine" value="amigo2"/>
+        </include>
+
+        <!-- PEOPLE RECOGNITION -->
```
Comment (PetervDooren) [inline launch/state_machines/challenge_storing_groceries.launch:29]: People recognition is not needed in storing groceries

File: launch/state_machines/gmapping.launch [hunk 6]
```diff
@@ -21,6 +23,17 @@
             <arg name="config" value="world_modeling/world_model_gmapping.yaml"/>
         </include>
 
+        <!-- IMAGE RECOGNITION -->
```
Comment (PetervDooren) [inline launch/state_machines/gmapping.launch:26]: Neither image nor people recognition is needed for gmapping

**Review by PetervDooren — DISMISSED** [review]

**Comment by MatthijsBurgh** [comment]
The question is, do you want to only launch a small part of the architecture, when not all parts are used? This means that the robot object is fully functional.

**Comment by PetervDooren** [comment]
That is a good question. We could make one launch file which launches most of the robots functionality. Then we would get something like the old world_model.launch.
On the other hand I think there is also merit in the 'dont use more than you need' mentality.
Maybe get some feedback from other team members on how to go about this.

**Comment by LoyVanBeek** [comment]
The image & people recognition stuff should preferably all run in 'trigger' or 'oneshot' mode and thus not consume much CPU when running in this way. 
Therefore, keeping the PR as it is is prefered, as it also reduces the [number of head scratches](http://globalnerdy.com/wordpress/wp-content/uploads/2008/02/wtfs_per_minute.jpg) when you do eventually need so image recognition  in a challenge and you can expect a fully functional robot. 

For the sake of speed, I say keep the stuff not core to the challenge.

**Comment by MatthijsBurgh** [comment]
Decided to run everything all the time, to have a fully functional robot object.

**Comment by MatthijsBurgh** [comment]
ping ;)

**Review by LoyVanBeek — APPROVED** [review]

**Pull request merged**